wgpu-hal/src/gles/queue.rs

```diff
@@ -808,34 +808,95 @@ impl super::Queue {
             C::EndQuery(target) => {
                 unsafe { gl.end_query(target) };
             }
+            C::TimestampQuery(query) => {
+                unsafe { gl.query_counter(query, glow::TIMESTAMP) };
+            }
             C::CopyQueryResults {
                 ref query_range,
                 ref dst,
                 dst_target,
                 dst_offset,
             } => {
-                self.temp_query_results.clear();
-                for &query in queries[query_range.start as usize..query_range.end as usize].iter() {
-                    let result = unsafe { gl.get_query_parameter_u32(query, glow::QUERY_RESULT) };
-                    self.temp_query_results.push(result as u64);
-                }
-                let query_data = unsafe {
-                    slice::from_raw_parts(
-                        self.temp_query_results.as_ptr() as *const u8,
-                        self.temp_query_results.len() * mem::size_of::<u64>(),
-                    )
-                };
-                match dst.raw {
-                    Some(buffer) => {
-                        unsafe { gl.bind_buffer(dst_target, Some(buffer)) };
+                if self
+                    .features
+                    .contains(wgt::Features::TIMESTAMP_QUERY_INSIDE_PASSES)
```

Comment:
I don't understand how this code works - we potentially need to deal with both occlusion and timestamp queries at all times?

Reply:
Similarly, both timestamps and occlusion queries should be able to use GL_QUERY_BUFFER if it's available, right?

Reply:
It might be better to make a private feature for query buffers.